recOrder/plugin/main_widget.py

```diff
@@ -1021,43 +1024,52 @@ def handle_lc_states_emit(self, value: tuple[tuple, dict[str, list]]):
                     textcoords="offset points",
                 )
 
-    @Slot(object)
-    def handle_bg_image_update(self, value):
+    def _add_or_update_image_layer(
+        self,
+        image: NDArray,
+        name: str,
+        cmap: str = "gray",
+        move_to_top: bool = True,
+    ):
+        """Add image layer of the given name if it does not exist, update existing layer otherwise.
 
-        if "Background Images" in self.viewer.layers:
-            self.viewer.layers["Background Images"].data = value
+        Parameters
+        ----------
+        image : NDArray
+            image intensity values
+        name : str
+            layer key name in napari layers list
+        cmap : str, optional
+            colormap to render in, by default "gray", use "rgb" for RGB images
+        move_to_top : bool, optional
+            whether to move the updated layer to the top of layers list, by default True
+        """
+        if name in self.viewer.layers:
+            self.viewer.layers[name].data = image
+            if move_to_top:
+                src_index = self.viewer.layers.index(name)
+                self.viewer.layers.move(src_index, dest_index=-1)
         else:
-            self.viewer.add_image(
-                value, name="Background Images", colormap="gray"
-            )
+            if cmap == "rgb":
+                self.viewer.add_image(image, name=name, rgb=True)
+            else:
+                self.viewer.add_image(image, name=name, colormap=cmap)
 
     @Slot(object)
-    def handle_bg_bire_image_update(self, value):
-
-        # Separate Background Retardance and Background Orientation
-        # Add new layer if none exists, otherwise update layer data
-        if "Background Retardance" in self.viewer.layers:
-            self.viewer.layers["Background Retardance"].data = value[0]
-        else:
-            self.viewer.add_image(
-                value[0], name="Background Retardance", colormap="gray"
-            )
+    def handle_bg_image_update(self, value):
+        self._add_or_update_image_layer(value, "Background Images")
 
-        if "Background Orientation" in self.viewer.layers:
-            self.viewer.layers["Background Orientation"].data = value[1]
-        else:
-            self.viewer.add_image(
-                value[1], name="Background Orientation", colormap="gray"
-            )
+    @Slot(object)
+    def handle_bg_bire_image_update(self, value):
+        self._add_or_update_image_layer(value[0], "Background Retardance")
+        self._add_or_update_image_layer(value[1], "Background Orientation")
```

Comment:
Possibly out of scope for this PR, but can we add `cmap="hsv"` to make the backgrounds consistent with acquisitions?

Reply:
Yes that makes sense. Will file new issue and PR.